daffodil-io/src/main/scala/org/apache/daffodil/io/DataOutputStreamImplMixin.scala

```diff
@@ -816,7 +816,7 @@ trait DataOutputStreamImplMixin extends DataStreamCommonState
   }
 
   final override def pastData(nBytesRequested: Int): ByteBuffer = {
-    Assert.usage(isReadable)
+    Assert.usage(isReadable || areDebugging)
     if (!areDebugging)
       throw new IllegalStateException("Must be debugging.")
```

Comment:
It's now impossible to hit this IllegalStateException. Note that the parse past/futureData functions do not have an assert but just have the if (!artDebugging) check. Should these all behave similarly, or are there reasons for the differences?